crates/sui-types/src/messages_checkpoint.rs

```diff
@@ -119,10 +124,137 @@ impl Default for ECMHLiveObjectSetDigest {
     }
 }
 
+/// CheckpointArtifact is a type that represents various artifacts of a checkpoint.
+/// We hash all the artifacts together to get the checkpoint artifacts digest
+/// that is included in the checkpoint summary.
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum CheckpointArtifact {
+    ObjectStates(Vec<ObjectRef>),
```

Comment:
Should we just leave this as a BTreeMap instead of making a temporary vector? Then we can pass `object_states.iter().map(|(id, (v, d))| (id, v, d))` to `build_from_unserialized`. Sorry for the nitpick but since this is a public enum, it seems better to make it impossible to construct an invalid state (i.e. an unsorted vector).